job/job.go

```diff
@@ -190,14 +192,18 @@ func (j *Job) UnmarshalJSON(b []byte) error {
 	}
 	j.DownloadTimeout = timeout
 
-	var useragent string
-	if useragentField, ok := tmp["user_agent"]; ok {
-		useragent, ok = useragentField.(string)
-		if !ok {
-			return errors.New("UserAgent must be a string")
+	rh, ok := tmp["request_headers"]
+	if ok {
+		if requestHeaders, ok := rh.(map[string]interface{}); ok {
+			headers := make(map[string]string)
+			for k, v := range requestHeaders {
+				headers[k] = v.(string)
```

Comment:
What if `v` cannot be casted to string? We should guard against this case and just ignore the header.

Reply:
Done, added an explanatory comment